ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -99,21 +100,30 @@ export class WorkflowLogsViewer extends React.Component<WorkflowLogsViewerProps,
         this.ensureUnsubscribed();
         this.subscription = services.workflows.getContainerLogs(this.props.workflow, this.props.nodeId, this.props.container, this.props.archived).subscribe(
             log => {
-                if (log) {
-                    this.setState(state => {
-                        log.split('\n').forEach(line => {
-                            state.lines.push(line);
-                        });
-                        return state;
-                    });
-                }
+                this.setState(state => {
+                    const newState = {...state, loaded: true};
+
+                    if (log) {
+                        newState.lines = newState.lines.concat(log.split('\n'));
+                    }
+
+                    return newState;
+                });
             },
             error => {
-                this.setState({error});
+                this.setState({error, loaded: true});
             }
         );
     }
 
+    private podHasNoLogs() {
```

Comment:
I'm using type inference here rather than explicitly specifying that the return type is `boolean` since typescript can infer that; is this acceptable here?